core(review): neutral review turns; add `[review].user_instructions` knob

kinopeee opened the pull request:
This PR refactors how user instructions are handled in review threads to preserve “neutral reviewer” isolation.

# Motivation
Previously, review threads inherited `user_instructions` from the parent session, effectively importing the full AGENTS.md stack into reviews. This violates the intended neutrality of the reviewer.

# What’s changed
- Neutral-by-default reviews
  - Review turns no longer inherit parent `user_instructions`.
- New config knob: `[review].user_instructions`
  - Narrowly scoped string, applied only to review threads when set.
  - Regular chat turns remain unchanged and continue to use AGENTS.md as before.
- Wiring & types
  - `ReviewToml { user_instructions: Option<String> }` (config types)
  - `ConfigToml { review: Option<ReviewToml> }`
  - `Config { review_user_instructions: Option<String> }`
  - `load_from_base_config_with_overrides` maps `cfg.review.user_instructions` to `Config.review_user_instructions`.
- Review turn behavior
  - In `spawn_review_thread`, set `TurnContext.user_instructions = config.review_user_instructions.clone()`.
- Tests
  - `review_respects_user_instructions` now sets `config.review_user_instructions = Some(...)` and keeps the 3‑item expectation (user instructions, environment context, review prompt) with `instructions == REVIEW_PROMPT`.
- Docs
  - Added a design note: `docs/review-user-instructions.md`.
  - Proposed a new “Review” section for `docs/config.md` (snippet included in a PR comment) documenting the knob and examples.

# Default behavior
- When `[review].user_instructions` is unset, review threads are neutral (no AGENTS.md imported).

# Usage
```toml
[review]
user_instructions = "Please respond concisely in Japanese."
```
CLI override example:
```sh
codex -c review.user_instructions='"Please respond concisely in Japanese."'
```

# Backward compatibility
- If anyone relied on AGENTS.md leaking into reviews, switch to the new knob and set only the minimal preference(s) you need (e.g., language).

# Verification
- Lint/format and tests run successfully locally for the core crate. Review tests updated to reflect the knob.

# Notes
- This PR intentionally does not alter non‑review turns. They continue to use AGENTS.md as usual.
- See the linked comment for the `docs/config.md` snippet to add under a new “Review” section.

ism-k commented:
@kinopeee 

Thanks for pushing on this! Speaking as another Japanese speaker, I’m excited to see review support get friendlier for language preferences. 

That said, wiring `user_instructions` to `parent_turn_context.user_instructions.clone()` would bring the entire `AGENTS.md` stack into review threads and undo the deliberate “neutral reviewer” isolation we rely on. All of the repo‑specific guidance (tone rules, workflow‑specific hacks, etc.) would start shaping the reviewer’s responses, which is exactly what we’re trying to avoid. Could we keep this field `None` and instead add a narrowly scoped mechanism—say a new `review.user_instructions` config knob—so we can pass just the small bits we actually need (like preferred language) without importing every `AGENTS.md` directive?

kinopeee committed:
core(review): neutral review turns; add [review].user_instructions knob

- Keep review turns neutral by default (no inherited user instructions)
- Add ReviewToml config type with user_instructions field
- Wire config.review_user_instructions in spawn_review_thread
- Add review_user_instructions field to Config struct
- Update run_task documentation to reflect neutral-by-default behavior
- Add test for review_respects_user_instructions

This allows users to pass narrowly scoped preferences (e.g., language)
to review threads without importing the full AGENTS.md stack.

kinopeee committed:
Refactor cloud tasks auth manager usage

kinopeee committed:
Merge branch 'openai:main' into fix-review-user-instructions

kinopeee commented:
## Additional fix: cloud-tasks auth configuration

While working on this PR, I also fixed a bug in `codex-rs/cloud-tasks` where the auth manager was always instantiated with `File` mode instead of respecting the user's `cli_auth_credentials_store` setting.

**Changes:**
- Added `util::load_cli_auth_manager()` helper that properly loads the user's config via `load_config_as_toml_with_cli_overrides`
- Updated `init_backend()` and `build_chatgpt_headers()` to use this helper
- Now respects `keyring`/`auto` settings correctly

**Files:**
- `codex-rs/cloud-tasks/src/lib.rs`
- `codex-rs/cloud-tasks/src/util.rs`

This fix addresses the issue where users with `cli_auth_credentials_store = "keyring"` would see "Not signed in" when running `codex cloud`.

Let me know if this should be split into a separate PR.

kinopeee committed:
Revert cloud-tasks changes (moved to separate PR #5856)

etraut-openai commented:
Thanks for the contribution, and apologies for the slow response. We've received many PRs, and we don't have the bandwidth on the codex team to review all of them.

We've updated our [contribution guidelines](https://github.com/openai/codex/blob/main/docs/contributing.md) to clarify that we're currently accepting contributions for bugs and security fixes, but we're not generally accepting new features at this time. We need to make sure that all new features compose well with both existing and upcoming features and fit into our roadmap. If you would like to propose a new feature, please file or upvote an enhancement request in the issue tracker. We will generally prioritize new features based on community feedback.
